maven-release-manager/src/main/java/org/apache/maven/shared/release/exec/InvokerMavenExecutor.java

```diff
@@ -113,7 +113,7 @@ public void executeGoals(
                 targetGoals.add(additionalArguments);
             }
 
-            req.setGoals(targetGoals);
+            req.addArgs(targetGoals);
```

Comment:
It is not a good idea. Each argument passed to `setGoals` are internal parsed, escaped and so on ...
 but `addArgs` pass arguments as is to command line process

look at line 113 `additionalArguments` is add as one item to list

Reply:
This is what the deprecation message says:

```
  /**
     * Sets the goals for the Maven invocation.
     *
     * @param goals The goals for the Maven invocation, may be <code>null</code> to execute the POMs default goal.
     * @return This invocation request.
     * @deprecated simply {@link #addArg(String)} or {@link #addArgs(Collection)} should be used
     */
```

Is that wrong then? Should this method be undeprecated?

Reply:
It is correctly deprecated ... but when we use new method we need new parameters where user set each command line argument separately

Parsing, escaping arguments are allwaye error prone, so we should required from users to provided it.

Reply:
Correct,  don't reinvent the wheel you will fail.